src/util/VariableNameCreator.ts

```diff
@@ -16,8 +16,84 @@ const reservedKeywords = [
 	"void",		  "volatile",	 "while",	"with",	"yield"
 ];
 const sapReservedKeywords = [ "sap" ];
+const reservedNativeTypes = [
+	"Array",
+	"ArrayBuffer",
+	"Atomics",
+	"BigInt",
+	"BigInt64Array",
+	"BigUint64Array",
+	"Boolean",
+	"DataView",
+	"Date",
+	"Error",
+	"EvalError",
+	"Float32Array",
+	"Float64Array",
+	"Function",
+	"Generator",
+	"GeneratorFunction",
+	"Infinity",
+	"Int16Array",
+	"Int32Array",
+	"Int8Array",
+	"InternalError",
+	"Intl",
+	"Intl.Collator",
+	"Intl.DateTimeFormat",
+	"Intl.Locale",
+	"Intl.NumberFormat",
+	"Intl.PluralRules",
+	"Intl.RelativeTimeFormat",
+	"JSON",
+	"Map",
+	"Math",
+	"NaN",
+	"Number",
+	"Object",
+	"Promise",
+	"Proxy",
+	"RangeError",
+	"ReferenceError",
+	"Reflect",
+	"RegExp",
+	"Set",
+	"SharedArrayBuffer",
+	"String",
+	"Symbol",
+	"SyntaxError",
+	"TypeError",
+	"TypedArray",
+	"URIError",
+	"Uint16Array",
+	"Uint32Array",
+	"Uint8Array",
+	"Uint8ClampedArray",
+	"WeakMap",
+	"WeakSet",
+	"WebAssembly",
+	"decodeURI",
+	"decodeURIComponent",
+	"encodeURI",
+	"encodeURIComponent",
+	"eval",
+	"globalThis",
+	"isFinite",
+	"isNaN",
+	"null",
+	"parseFloat",
+	"parseInt"
```

Comment:
also have a looks at:
`Object.keys(window).join(", ")`

Reply:
I would rather rely on https://github.com/sindresorhus/globals .

Or, check k against the globals in each module. If `UInt8Array` is not used, why search for  another name?

Reply:
+1 for using the `globals` package

Reply:
that's a neat module :) covered with c7c02ec